src/relations/mysql.py

```diff
@@ -57,32 +66,63 @@ def _get_or_set_password_in_peer_databag(self, username: str) -> str:
     def _on_leader_elected(self, _) -> None:
         """Handle the leader elected event.
 
-        Retrieves relation data from the peer relation databag and copies
-        the relation data into the new leader unit's databag.
+        Updates a key in the peer relation databag in order to trigger the peer
+        relation changed event.
         """
         # Skip if the charm is not past the setup phase (config-changed event not executed yet)
         if not self.charm._is_peer_data_set:
             return
 
-        relation_data = json.loads(self.charm.app_peer_data.get("mysql_relation_data", "{}"))
+        # Trigger a peer relation changed event in order to refresh the relation data
+        leader_elected_count = int(self.charm.app_peer_data.get("leader_elected_count", "1"))
+        self.charm.app_peer_data["leader_elected_count"] = str(leader_elected_count + 1)
```

Comment:
I would expect that it's ok to default to 1 given that:
a. most probably, the first leader elected event wont be catch here, due it happening before peer_data being seted.
b. most important, we don't really care, since this is just a mechanism to trigger a `peer_relation_changed` event

Reply:
paulo captured my thinking precisely. the `leader-elected` in its first invocation sets the peer data (which is why i defaulted to 1 to set the value accurately). additionally, it's meant as a mechanism to to trigger the `peer_relation_changed` event